homeassistant/components/google_pubsub/__init__.py

```diff
@@ -45,9 +45,12 @@ def setup(hass: HomeAssistant, yaml_config: dict[str, Any]):
     config = yaml_config[DOMAIN]
     project_id = config[CONF_PROJECT_ID]
     topic_name = config[CONF_TOPIC_NAME]
-    service_principal_path = os.path.join(
-        hass.config.config_dir, config[CONF_SERVICE_PRINCIPAL]
-    )
+    if hass.config.config_dir:
+        service_principal_path = os.path.join(
+            hass.config.config_dir, config[CONF_SERVICE_PRINCIPAL]
+        )
+    else:
+        service_principal_path = os.path.join(config[CONF_SERVICE_PRINCIPAL])
```

Comment:
What does this join? 😬

Reply:
It joins config[CONF_SERVICE_PRINCIPAL] to the existing os.path.

I cannot do os.path(config[CONF_SERVICE_PRINCIPAL]))

Reply:
![image](https://user-images.githubusercontent.com/195327/129633559-55f78e5b-6d3e-495f-b15b-bdab10a8535d.png)

Reply:
Done.